packages/admin/admin/src/common/buttons/CopyToClipboardButton.tsx

```diff
@@ -1,47 +1,53 @@
 import { Accept, Copy } from "@comet/admin-icons";
-import { ComponentsOverrides, Grow, IconButton, SvgIconProps, Theme } from "@mui/material";
-import { IconButtonProps } from "@mui/material/IconButton/IconButton";
-import { createStyles, WithStyles, withStyles } from "@mui/styles";
-import clsx from "clsx";
+import { ComponentsOverrides, css, Grow, IconButton, styled, Theme, useThemeProps } from "@mui/material";
+import { ThemedComponentBaseProps } from "helpers/ThemedComponentBaseProps";
 import * as React from "react";
 
-export interface CopyToClipboardButtonComponents {
-    CopyIcon?: React.ElementType;
-    SuccessIcon?: React.ElementType;
-}
-
-export interface CopyToClipboardButtonComponentsProps {
-    CopyButton?: Partial<IconButtonProps>;
-    successButton?: Partial<IconButtonProps>;
-    copyIcon?: Partial<SvgIconProps>;
-    successIcon?: Partial<SvgIconProps>;
-}
-
-export interface CopyToClipboardButtonProps {
+export interface CopyToClipboardButtonProps
+    extends ThemedComponentBaseProps<{
+        root: "div";
+        buttonContainer: "div";
+        copyButtonContainer: "div";
+        successButtonContainer: "div";
+        button: "div";
+        copyButton: typeof IconButton;
+        successButton: typeof IconButton;
+    }> {
     copyText: string;
-    components?: CopyToClipboardButtonComponents;
-    componentsProps?: CopyToClipboardButtonComponentsProps;
+    copyIcon?: React.ReactNode;
+    successIcon?: React.ReactNode;
 }
 
-const CopyToClipboardButton = ({
-    classes,
-    copyText,
-    components = {},
-    componentsProps = {},
-}: CopyToClipboardButtonProps & WithStyles<typeof styles>): React.ReactElement => {
-    const { CopyIcon = Copy, SuccessIcon = Accept } = components;
+export type CopyToClipboardButtonClassKey =
+    | "root"
+    | "showSuccess"
+    | "buttonContainer"
+    | "copyButtonContainer"
+    | "successButtonContainer"
+    | "button"
+    | "copyButton"
+    | "successButton";
 
+type OwnerState = {
+    showSuccess: boolean | undefined;
```

Comment:
I don't think this should ever be `undefined`

```suggestion
    showSuccess: boolean;
```